docker_build/Dockerfile

```diff
@@ -0,0 +1,80 @@
+# An integration test & dev container which builds and installs cuDF from main
+ARG CUDA_VERSION=10.2
+ARG CUDA_SHORT_VERSION=${CUDA_VERSION}
+
+FROM nvidia/cuda:${CUDA_VERSION}-devel
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/lib
+# Needed for cudf.concat(), avoids "OSError: library nvvm not found"
+ENV NUMBAPRO_NVVM=/usr/local/cuda/nvvm/lib64/libnvvm.so
+ENV NUMBAPRO_LIBDEVICE=/usr/local/cuda/nvvm/libdevice/
+ENV DEBIAN_FRONTEND=noninteractive
+
+ENV CC=/usr/bin/gcc
+ENV CXX=/usr/bin/g++
+
+# Build env variables for arrow
+ENV CMAKE_BUILD_TYPE=release
+ENV PYARROW_WITH_PARQUET=1
+ENV PYARROW_WITH_CUDA=1
+ENV PYARROW_WITH_ORC=1
+ENV PYARROW_WITH_DATASET=1
+
+ENV ARROW_HOME=/repos/dist
+ENV LD_LIBRARY_PATH=/repos/dist/lib:$LD_LIBRARY_PATH
+
+# Build env variables for rmm
+ENV INSTALL_PREFIX=/usr
+
+
+RUN apt update -y --fix-missing && \
+    apt upgrade -y && \
+    apt install -y --no-install-recommends \
+      git \
+      libboost-all-dev \
+      python3.7-dev \
+      build-essential \
+      cmake \
+      software-properties-common \
+      llvm-7 \
+      autoconf \
+      bison \
+      flex \
+      libboost-filesystem-dev \
+      libboost-system-dev \
+      libboost-regex-dev \
+      libjemalloc-dev \
+      wget \
+      libssl-dev \
+      protobuf-compiler \
+      clang && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* && \
+    add-apt-repository ppa:deadsnakes/ppa && \
+    update-alternatives --install /usr/bin/python python /usr/bin/python3.7 1 && \
+    apt-get install curl && \
+    curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
+    python3.7 get-pip.py
+
+# Install llvm 11.1 for apache
+RUN wget https://github.com/llvm/llvm-project/releases/download/llvmorg-11.1.0/clang+llvm-11.1.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz && tar -xvf clang+llvm-11.1.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
+    mv clang+llvm-11.1.0-x86_64-linux-gnu-ubuntu-16.04 llvm-11.1 && cp -R llvm-11.1 /usr/lib/ && rm -rf clang+llvm-11.1.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz && \
+    wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && \
+    apt-get install llvm-10-dev llvm-9-dev -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
Does Arrow not have a mechanism to handle this automatically as part of the Arrow build? This is kinda gross

Reply:
Disabled building `gandiva` for arrow, we'd have to install multiple llvm versions manually if we want to build it and arrow doesn't seem to be handling it automatically.